business_requirement/models/business.py

```diff
@@ -0,0 +1,436 @@
+        copy=False,
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    parent_id = fields.Many2one(
+        comodel_name='business.requirement',
+        string='Parent',
+        ondelete='set null',
+        domain="[('id', '!=', id)]",
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    level = fields.Integer(
+        compute='_get_level',
+        string='Level',
+        store=True
+    )
+    change_request = fields.Boolean(
+        string='Change Request?',
+        default=False,
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    project_id = fields.Many2one(
+        comodel_name='project.project',
+        string='Master Project',
+        ondelete='set null',
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    partner_id = fields.Many2one(
+        comodel_name='res.partner',
+        string='Customer',
+        store=True,
+    )
+    sub_br_count = fields.Integer(
+        string='Count',
+        compute='_sub_br_count'
+    )
+    priority = fields.Selection(
+        [('0', 'Low'), ('1', 'Normal'), ('2', 'High')],
+        'Priority',
+        required=True,
+        default='1'
+    )
+    confirmation_date = fields.Datetime(
+        string='Confirmation Date',
+        readonly=True
+    )
+    confirmed_id = fields.Many2one(
+        'res.users', string='Confirmed by',
+        readonly=True
+    )
+    approval_date = fields.Datetime(
+        string='Approval Date',
+        readonly=True
+    )
+    approved_id = fields.Many2one(
+        'res.users',
+        string='Approved by',
+        readonly=True
+    )
+    company_id = fields.Many2one(
+        'res.company', string='Company',
+        required=True, readonly=True, states={'draft': [('readonly', False)]},
+        default=_get_default_company,
+    )
+
+    @api.one
+    @api.onchange('project_id')
+    def project_id_change(self):
+        if self.project_id and self.project_id.partner_id:
+            self.partner_id = self.project_id.partner_id.id
+
+    @api.model
+    def create(self, vals):
+        if vals.get('name', '/') == '/':
+            vals['name'] = self.env['ir.sequence'].get('business.requirement')
+        return super(BusinessRequirement, self).create(vals)
+
+    @api.multi
+    @api.depends('parent_id')
+    def _get_level(self):
+        def _compute_level(br):
+            return br.parent_id and br.parent_id.level + 1 or 1
+
+        for br in self:
+            level = _compute_level(br)
+            br.level = level
+
+    @api.one
+    @api.depends('business_requirement_ids')
+    def _sub_br_count(self):
+        self.sub_br_count = len(self.business_requirement_ids)
+
+    @api.model
+    def _get_states(self):
+        states = [
+            ('draft', 'Draft'),
+            ('confirmed', 'Confirmed'),
+            ('approved', 'Approved'),
+            ('customer_approval', 'Customer Approval'),
```

Comment:
I wonder whether we should be more generic as "customer approval" is for outside projects.
something like "Internal Approval" and "Stakeholder Approval"